hyperspy/roi.py

```diff
@@ -628,7 +641,13 @@ def _apply_roi2widget(self, widget):
         widget.set_bounds(left=self.left, right=self.right)
 
     def _get_widget_type(self, axes, signal):
-        return widgets.RangeWidget
+        direction = guess_vertical_or_horizontal(axes=axes, signal=signal)
```

Comment:
If the return of the `guess_vertical_or_horizontal` function is changed appropriately as mentioned above, this could be simplified to:
```python
return partial(widgets.RangeWidget, direction=guess_vertical_or_horizontal(axes=axes, signal=signal))
```

Reply:
For the same reason as above, I would prefer to leave it as-is.